src/java/picard/illumina/IlluminaBasecallsToFastq.java

```diff
@@ -68,15 +68,32 @@
 import java.util.Set;
 
 @CommandLineProgramProperties(
-        usage = "Generate fastq file(s) from data in an Illumina basecalls output directory.\n" +
-                "Separate fastq file(s) are created for each template read, and for each barcode read, in the basecalls.\n" +
-                "Template fastqs have extensions like .<number>.fastq, where <number> is the number of the template read,\n" +
-                "starting with 1.  Barcode fastqs have extensions like .barcode_<number>.fastq, where <number> is the number\n" +
-                "of the barcode read, starting with 1.",
-        usageShort = "Generate fastq file(s) from data in an Illumina basecalls output directory",
-        programGroup = Illumina.class
+  usage = IlluminaBasecallsToFastq.USAGE_SUMMARY + IlluminaBasecallsToFastq.USAGE_DETAILS,
+  usageShort = IlluminaBasecallsToFastq.USAGE_SUMMARY,
+  programGroup = Illumina.class
 )
 public class IlluminaBasecallsToFastq extends CommandLineProgram {
+  static final String USAGE_SUMMARY = "Generate fastq file(s) from Illumina basecall read data.  ";
+  static final String USAGE_DETAILS = "<p>This tool generates fastq files from data in an Illumina basecalls (*.bcl) output directory.  " +
+          "Separate fastq files are created for each template, barcode, and index read.   In the absence of sample pooling (multiplexing) and/or barcodes, then an OUTPUT_PREFIX (file) must be provided as the sample identifier.  For multiplexed samples, a MULTIPLEX_PARAMS (file) must be used.  The MULTIPLEX_PARAMS file contains the list of sample barcodes used to sort template, barcode, and index reads.  It is essentially the same as the BARCODE_FILE used in the" +
```

Comment:
if you put  "string" that has a directory in it, it might be less
confusing...e.g.

OUTPUT_PREFIX=/user/home/gatk/experiments/noBarcode.1

On Tue, May 24, 2016 at 7:19 PM, David E. Kling <notifications@github.com>
wrote:

> In src/java/picard/illumina/IlluminaBasecallsToFastq.java
> <https://github.com/broadinstitute/picard/pull/521#discussion_r64492144>:
>
> >  )
> >  public class IlluminaBasecallsToFastq extends CommandLineProgram {
> > +  static final String USAGE_SUMMARY = "Generate fastq file(s) from Illumina basecall read data.  ";
> > +  static final String USAGE_DETAILS = "<p>This tool generates fastq files from data in an Illumina basecalls (*.bcl) output directory.  " +
> > +          "Separate fastq files are created for each template, barcode, and index read.   In the absence of sample pooling (multiplexing) and/or barcodes, then an OUTPUT_PREFIX (file) must be provided as the sample identifier.  For multiplexed samples, a MULTIPLEX_PARAMS (file) must be used.  The MULTIPLEX_PARAMS file contains the list of sample barcodes used to sort template, barcode, and index reads.  It is essentially the same as the BARCODE_FILE used in the" +
>
> I didn't make a code change. As per @yfarjoun
> <https://github.com/yfarjoun>'s suggestion, I just changed the example to
> have a string e.g. [OUTPUT_PREFIX=noBarcode.1] and not a file name
> [OUTPUT_PREFIX=noBarcode.1.fastq] even though it is declared as a public
> file.
>
> I have to say, I am still confused by this.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/521/files/eb30e0911b5269b06a336470afc1d258b54283e2#r64492144>
>